clang/lib/Sema/SemaDecl.cpp

```diff
@@ -20008,6 +20008,87 @@ bool Sema::IsValueInFlagEnum(const EnumDecl *ED, const llvm::APInt &Val,
   return !(FlagMask & Val) || (AllowMask && !(FlagMask & ~Val));
 }
 
+bool Sema::ComputeBestEnumProperties(ASTContext &Context, EnumDecl *Enum,
+                                     bool is_cpp, bool isPacked,
```

Comment:
Unless I'm missing something, `Enum` parameter is actually unused.
Also, `is_cpp` can be queried from `ASTContext`, via `Context.getLangOpts().CPlusPlus`.
`Context` parameter is not required, since it can be queried from `Sema` itself via simply using `Context`.
I would expect that `BestWidth` is width of the `BestType`, right? It probably can be simply queried via `ASTContext::getIntWidth(BestType)` instead of passing to/from a function.

Reply:
Removed `Enum`, `is_cpp` and `BestWidth`, but unfortunately when calling this function from LLDB, `Context` in Sema is empty, so it has to be passed as an argument.

Reply:
Hmm, then I wonder if it makes sense to make it a method of ASTContext instead of method of Sema? Usually ASTContext is used to query type information, so it might make sense.

Reply:
Maybe better to put this function into `EnumDecl` itself? It has `ASTContext` within it filled on both Sema and LLDB side, and then `EnumDecl`  can fill it's own `BestType` and `BestPromotionType` fields using this function.